applications/frontier/src/frontier_handle_rate.erl

```diff
@@ -0,0 +1,330 @@
+    lager:debug("Handle rate limits request for ~s", [Entity]),
+    Limits = lookup_rate_limit_records(Entity, IncludeRealm, MethodList),
+    send_response(Limits, JObj).
+
+-spec lookup_methods(wh_json:object()) -> api_binaries().
+lookup_methods(JObj) ->
+    Method = wh_json:get_value(<<"Method">>, JObj),
+    MethodName = case Method of
+                     'undefined' -> 'undefuned';
+                     _ -> [resolve_method(Method)]
+                 end,
+    Methods = wh_json:get_value(<<"Method-List">>, JObj),
+    MethodNames = case is_list(Methods) of
+                      'true' -> lists:map(fun resolve_method/1, Methods);
+                      'false' -> []
+                  end,
+    case is_list(MethodName) of
+        'false' -> MethodNames;
+        'true' -> MethodName
+    end.
+
+-spec send_response(wh_json:object(), wh_json:object()) -> 'ok'.
+send_response(Limits, Reqest) ->
+    RespStub = wh_json:from_list([{<<"Msg-ID">>, wh_json:get_value(<<"Msg-ID">>, Reqest)}
+                                   | wh_api:default_headers(?APP_NAME, ?APP_VERSION)
+                                  ]),
+    Srv = wh_json:get_value(<<"Server-ID">>, Reqest),
+    wapi_frontier:publish_ratelimits_resp(Srv, wh_json:merge_jobjs(Limits, RespStub)).
+
+
+-spec lookup_rate_limit_records(ne_binary()) -> wh_json:object().
+lookup_rate_limit_records(Entity) ->
+    lookup_rate_limit_records(Entity, 'true', names()).
+
+-spec lookup_rate_limit_records(ne_binary(), boolean(), ne_binaries()) -> wh_json:object().
+lookup_rate_limit_records(Entity, IncludeRealm, MethodList) ->
+    lager:info("Handle rate limit request for ~s", [Entity]),
+    Realm = frontier_utils:extract_realm(Entity),
+    Responses = case whapps_util:get_account_by_realm(Realm) of
+                    {'ok', AccountDB} ->
+                        lager:info("Found realm, try to send response"),
+                        run_rate_limits_query(Entity, AccountDB, IncludeRealm, MethodList);
+                    _ ->
+                        lager:info("Can't find realm ~s. Throttle him.", [Realm]),
+                        make_deny_rates(Entity, IncludeRealm, MethodList)
+                end,
+    lists:foldl(fun fold_responses/2, wh_json:new(), lists:flatten(Responses)).
+
+-spec build_list_of_querynames(ne_binary(), boolean()) -> ne_binaries().
+build_list_of_querynames(Entity, IncludeRealm) ->
+    Realm = frontier_utils:extract_realm(Entity),
+    Username = frontier_utils:extract_username(Entity),
+    QueryName = case Realm =/= Entity of
+                    'true' -> Username;
+                    'false' -> Realm
+                end,
+    case IncludeRealm andalso Realm =/= Entity of
+        'true' -> [Realm, Username];
+        'false' -> [QueryName]
+    end.
+
+
+-spec run_rate_limits_query(ne_binary(), ne_binary(), boolean(), ne_binary()) -> wh_json:objects().
+run_rate_limits_query(Entity, AccountDB, IncludeRealm, MethodList) ->
```

Comment:
No queries are being run, that I can see. Perhaps this should be named something more like 'build_rates' or something more description?